core/domain/classifier_services.py

```diff
@@ -224,3 +226,117 @@ def delete_classifier(classifier_id):
     classifier_data_model = classifier_models.ClassifierDataModel.get(
         classifier_id)
     classifier_data_model.delete()
+
+
+def get_classifier_training_job_from_model(classifier_training_job_model):
+    """Gets a classifier training job domain object from a classifier
+       training job model.
+
+    Args:
+        classifier_training_job_model: Classifier training job instance in
+            datastore.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+    """
+    return classifier_domain.ClassifierTrainingJob(
+        classifier_training_job_model.id,
+        classifier_training_job_model.algorithm_id,
+        classifier_training_job_model.exp_id,
+        classifier_training_job_model.exp_version_when_created,
+        classifier_training_job_model.state_name,
+        classifier_training_job_model.status,
+        classifier_training_job_model.training_data)
+
+def get_classifier_training_job_by_id(job_id):
+    """Gets a classifier training job from a job_id.
+
+    Args:
+        job_id: str. ID of the classifier training job.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Raises:
+        Exception: Entity for class ClassifierTrainingJobModel with id not
+            found.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id))
+    classifier_training_job = get_classifier_training_job_from_model(
+        classifier_training_job_model)
+    return classifier_training_job
+
+
+def _create_classifier_training_job(classifier_training_job):
+    """Creates classifier training job model in the datastore given a
+       classifier training job domain object.
+
+    Args:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    job_id = classifier_models.ClassifierTrainingJobModel.create(
+        classifier_training_job.algorithm_id, classifier_training_job.exp_id,
+        classifier_training_job.exp_version_when_created,
```

Comment:
_If the re-training function holds True, we create another job instance. The exp_version_when_created is now updated to the latest version of the exploration._

Umm no, we would create a new classifier altogether in this case because we would like to keep the old classifiers around for sometime.

@seanlip ah, I got what you meant now! I thought about this for sometime and at one point I agreed with @pranavsid98. But you are right, we used the terminology "exp_version_when_created" so as to explicitly say that this classifier can span over multiple exp_versions but there is no use being explicit here because training_job is created just for the current exp_version.